From b595cb0d5355dcad208ebd0a9469676ab0133aab Mon Sep 17 00:00:00 2001
From: Mike Christie <michaelc@cs.wisc.edu>
Date: Tue, 25 Aug 2015 22:13:39 -0500
Subject: [PATCH] iscsid: fix iscsi_host_set_net_params return code

iscsi_host_set_net_params was returning ISCSI_ERR and EXYZ error
codes. It should return the ISCSI_ERR type.
---
 usr/initiator_common.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/usr/initiator_common.c b/usr/initiator_common.c
index 3053812..04f8d20 100644
--- a/usr/initiator_common.c
+++ b/usr/initiator_common.c
@@ -261,7 +261,7 @@ int iscsi_setup_portal(struct iscsi_conn *conn, char *address, int port)
 	return 0;
 }
 
-int host_set_param(struct iscsi_transport *t,
+static int host_set_param(struct iscsi_transport *t,
 		   uint32_t host_no, int param, char *value,
 		   int type)
 {
@@ -273,7 +273,7 @@ int host_set_param(struct iscsi_transport *t,
 		log_error("can't set operational parameter %d for "
 			  "host %d, retcode %d (%d)", param, host_no,
 			  rc, errno);
-		return rc;
+		return ISCSI_ERR_INVAL;
 	}
 	return 0;
 }
@@ -634,7 +634,7 @@ int iscsi_host_set_net_params(struct iface_rec *iface,
 	if (!iface_is_bound_by_ipaddr(iface)) {
 		log_warning("Please set the iface.ipaddress for iface %s, "
 			    "then retry the login command.\n", iface->name);
-		return EINVAL;
+		return ISCSI_ERR_INVAL;
 	}
 
 	/* these type of drivers need the netdev upd */
-- 
2.7.4

